### PR TITLE
Log KL Divergence in GRPO Loss function

### LINKED ISSUE #298: Log KL Divergence in GRPO Loss function

### PR DESCRIPTION
**Pull request by krammnic**
Resolves #298

### CODE REVIEW

File: src/forge/actors/trainer.py [hunk 1]
```diff
@@ -179,7 +181,7 @@ async def setup(self):
 
     def forward_backward(
         self, inputs: dict[str, Tensor], targets: dict[str, Tensor]
-    ) -> Tensor:
+    ) -> Tensor | LossMetrics:
```
Comment (casteryh) [inline src/forge/actors/trainer.py:184]: shouldn't this be `tuple[..,..]`?

Comment (krammnic) [reply src/forge/actors/trainer.py:184]: It should!

**Comment by krammnic** [comment]
@casteryh let's merge

File: src/forge/losses/grpo_loss.py [hunk 2]
```diff
@@ -7,23 +7,26 @@
...
         https://github.com/huggingface/trl/blob/417915a3e4d3e3bc8d7b196594308b8eabf928be/trl/trainer/grpo_trainer.py#L1624.
     """
 
-    def __init__(self, beta: float = 0.1):
+    def __init__(self, beta: float = 0.1) -> tuple[torch.Tensor, LossMetrics]:
         super().__init__()
         self.beta = beta
 
     def forward(self, logprobs, ref_logprobs, advantages, padding_mask):
         kl = torch.exp(ref_logprobs - logprobs) - (ref_logprobs - logprobs) - 1
```
Comment (joecummings) [inline src/forge/losses/grpo_loss.py:26]: Can we log the KL divergence minus padding tokens? May have to move that op up in the loss function.

File: src/forge/losses/grpo_loss.py [hunk 3]
```diff
@@ -7,23 +7,26 @@
 import torch
 from torch import nn
 
+from forge.data_models.loss_metrics import LossMetrics
```
Comment (joecummings) [inline src/forge/losses/grpo_loss.py:10]: I'm not sure this is a fully fleshed out data model we want to use. 

For now could we just define a loose type in this file and shove the metrics in that?

Comment (krammnic) [reply src/forge/losses/grpo_loss.py:10]: I've done it with data_model, because we might want to log some other things from different losses in future (margins from DPO loss for instance).

File: src/forge/losses/grpo_loss.py (hunk 2 above)
Comment (krammnic) [reply src/forge/losses/grpo_loss.py:26]: Yep good idea

Comment (krammnic) [reply src/forge/losses/grpo_loss.py:26]: Adressed

**Comment by joecummings** [comment]
We landed this !
